source/fundamentals/data-formats/bson.txt

```diff
@@ -0,0 +1,224 @@
+.. _csharp-bson:
+
+==============
+Work with BSON
+==============
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to create BSON documents, read BSON from a file, 
+and write BSON to a file using the {+driver-short+}.
+
+BSON Data Format
+----------------
+
+**BSON**, or Binary JSON, is the data format that MongoDB uses to organize
+and store data. This data format includes all JSON data structure types and
+adds support for types including dates, different size integers, ObjectIds, and
+binary data. For a complete list of supported types, see the
+:manual:`BSON Types </reference/bson-types>` server manual page.
+
+The code samples in this guide use the following BSON document as an example:
+
+.. code-block:: none
+
+   {
+       "address" : {
+           "street" : "Pizza St", 
+           "zipcode" : "10003" 
+       },
+       "coord" : [-73.982419, 41.579505]
+       "cuisine" : "Pizza",
+       "name" : "Mongo's Pizza"
+   }
+
+Create a BSON Document
+----------------------
+
+To build a BSON document in {+language+}, create an instance of the ``BsonDocument`` class.
+The ``BsonDocument`` constructor accepts ``BsonElement`` arguments that map to the fields
+and values in the document. You can either construct individual ``BsonElement`` objects 
+and pass them to the ``BsonDocument`` constructor or pass field-value pairs inside curly 
```

Comment:
Would it be worthwhile to show an example of passing ``BsonElement`` objects into the constructor?

Reply:
That's a good idea, as long as it doesn't make the example too confusing. Let me know what you think.

Reply:
Hmmm seeing it together I think that could be more confusing. Could we split it into two examples? So instead of saying "You can either do X or do Y" (like it says now), it could say:

> "You can pass by field value pairs...."
> Example
> "You can also pass BSONElements to the constructor...."
> Example

(I switched the ordering so that the last example shown is the one used through the rest of the page)